### DISCUSSION TITLE
Are spaces allowed between " :" and "param" in the field attribute?

### DISCUSSION THREAD

**Discussion by yuki-kimoto** [Ovid/Cor#70] [Q&A] [answered]
I'm seeing the field grammer. I find an ambiguous expression.

https://github.com/Ovid/Cor/blob/master/rfc/grammar.md#33-field-grammar

```
ATTRIBUTE        ::= ':' (
                              'param'     NAME?   # allowed in constructor
                            | 'name'      NAME?   # alternate name (defaults to field name minus the sigil)
                            | 'reader'    NAME?   # $field method to read the field
                            | 'writer'    NAME?   # set_$field method to write the field
                            | 'predicate' NAME?   # is_$field method to test if field is defined
                            | 'common'            # identifies field as class method
                            | HANDLES
                     )

...

PAIR             ::= IDENTIFIER  ':' IDENTIFIER

```

Are spaces allowed between " :" and "param", etc in the field attribute?

```
:    param
```

**Comment by Ovid** [comment, marked as answer]
Yes. These are normal [attributes](https://metacpan.org/pod/attributes), with the argument to the attribute receiving special handling by @leonerd's code.

**Comment by yuki-kimoto** [comment]
The link's attributes allow the following code <code> : MyAttribute Test</code> .

https://metacpan.org/pod/attributes

```
sub foo : MyAttribute Test;
```

On the other hand, In Corrina, it looks like <code>:param</code> and <code>:reader</code>

```
field $x :param :reader;
```

Dose Corrina allow the following code?

```
field $x : param reader;
```

**Reply by Ovid** [reply]
@leonerd Might be better prepared to answer that one. I actually forgot about that syntax. To be honest, I would prefer an enforcement of the leading colon, no space, for all attributes, but that's not my call to make (and that might just be an aesthetic preference).

**Reply by yuki-kimoto** [reply]
Thanks. I wait for the preparation.

**Reply by rwp0** [reply]
That's a good catch. Syntax-wise that's really interesting to know.

**Reply by tobyink** [reply]
Perl is generally pretty liberal about allowing extra whitespace.

```shell
perl -E'my $   x = 42; say $ x'
```
